reports/views.py

```diff
@@ -0,0 +1,155 @@
+from datetime import timedelta
+
+from django.contrib.auth.decorators import login_required
+from django.http import FileResponse, Http404
+from django.shortcuts import get_object_or_404, render
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.utils import timezone
+from knox.models import AuthToken
+from rest_framework.decorators import api_view
+
+from permits import services
+from permits.decorators import permanent_user_required
+from permits.models import ComplementaryDocumentType, WorksObjectType
+from permits.serializers import PermitRequestPrintSerializer
+
+from .models import Report
+from .utils import run_docker_container
+
+
+def user_is_allowed_to_generate_report(
+    request, permit_request_id, work_object_type_id, report_id
+):
+
+    # Check that user has permission to generate pdf
+    if not (
+        request.user.has_perm("reports.can_generate_pdf") or request.user.is_superuser
+    ):
+        raise Http404
+
+    # Check that user is allowed to see the current permit_request
+    permit_request = get_object_or_404(
+        services.get_permit_requests_list_for_user(request.user), pk=permit_request_id
+    )
+    # Check the current work_object_type_id is allowed for user
+    # The wot list is associated to a permitrequest, thus a user that have
+    # access to the permitrequest, also has access to all wots associated with it
+    if not (
+        work_object_type_id in permit_request.wot_id_list() or request.user.is_superuser
```

Comment:
even if user is super-user, he shouldn't really be able to print a report for a wot that is not part of the permit request, no ?
(not for security, but it doesn't make sense)

Reply:
Yes that's right. Done